doc/subsystems/test/ztest.rst

```diff
@@ -82,6 +82,55 @@ The above test is then enabled as part of the testsuite using::
     ztest_unit_test(test_assert)
 
 
+Listing Tests
+=============
+
+Tests (test projects) in the Zephyr tree consist of many testcases that run as
+part of a project and test similar functionality, for example an API or a
+feature. The ``sanitycheck`` script can parse the the testcases in all
+test project or a subset of them and can generate reports on a more granular
+level, i.e. if cases have passed or failed or if they were blocked or skipped.
+
+To know about skipped or blocked test cases we do parse the case names from the
+source files and keep track of them while running the tests. To list all kernel
+cases for example, enter::
+
+        sanitycheck --list-tests -T tests/kernel
+
+Skipping Tests
+==============
+
+In many cases sepcial or architecture specific tests will not be run on all
```

Comment:
Add some punctuation and make shorter sentences.  How about:

    Special- or architecture-specific tests cannot run on all
    platforms and architectures, however we still want to count those and
    report them as being skipped.  Because the test inventory and
    the list of tests is extracted from the code, adding
    conditionals inside the test suite is sub-optimal.  Tests that need
    to be skipped for a certain platform or feature need to explicitly
    report a skip using :c:func:`ztest_test_skip()`. If the test runs,
    it needs to report either a pass or fail.  For example::